app/src/main/java/com/shopmate/shopmate/AddItemActivity.java

```diff
@@ -184,4 +215,82 @@ private void requestWritePermission() {
         }
     }
 
+    //Ties all of the information passed from the client to the backend server via ShopMateService API
+    private void syncItem() throws ExecutionException, InterruptedException{
+
+        //Get the list name & list id values from the ShoppingListActivity
+        Intent intent = getIntent();
+        Bundle extras = intent.getExtras();
+        String listName = extras.getString("title");
+        int listId = Integer.parseInt(extras.getString("listId"));
+
+        // Get the ShopMateService singleton
+        ShopMateService service = ShopMateServiceProvider.get();
+
+        // Get the user's Facebook token
+        String fbToken = AccessToken.getCurrentAccessToken().getToken();
+
+        //Get user data from the screen
+        EditText itemDescription = (EditText)findViewById(R.id.itemDesc);
+        EditText itemPrice = (EditText)findViewById(R.id.itemPrice);
+        EditText itemQuantity = (EditText)findViewById(R.id.itemQty);
+
+
+        //Create a shopping list item to be used in the next API call
+        ShoppingListItem testItem = new ShoppingListItemBuilder(listName)
+                .description(itemDescription.getText().toString().trim())
+                .imageUrl(selectedImageUri == null ? "" : selectedImageUri.toString())
```

Comment:
Only call this if a URL is actually set. Also, using selectedImageUri isn't exactly valid because this has to be an HTTP(S) URL.